src/Types/Node.php

```diff
@@ -62,4 +65,34 @@ public function id(): int
     {
         return $this->id;
     }
+
+    public function property(string $key)
+    {
+        if ($this->properties->hasKey($key)) {
+            return $this->properties->get($key);
+        }
+    }
+
+    private function mapProperties(Map $properties): Map
+    {
+        return $properties->map(function($key, $value) {
+            if (is_object($value)) {
+                switch(get_class($value)) {
+                    case BoltDate::class:
+                        return Date::makeFromBoltDate($value);
+                    case BoltDuration::class:
+                        return Duration::makeFromBoltDuration($value);
+                    case BoltDateTime::class:
+                        return DateTime::makeFromBoltDateTime($value);
+                }
+            }
+
+            return $value;
+        });
+    }
+
+    public function __get($key)
+    {
+        return $this->property($key);
+    }
```

Comment:
@transistive I may have taken a step out of context here so i would like your thoughts on this. Whether it is within the scope of this driver or not to be able to access Node properties as attributes.

```php
$article = $result['article'];
$article->created;
$article->published_at;
```

Instead of having to go through properties manually or calling `Node::property('created')`, this acts as a magic decoration.

Reply:
Implementing `__set` is a good idea 👍 

My intention is to keep "magic" to a minimum and so far it looks like we won't be needing much more than this so it's preferable to go about it case-by-case.